2017/11/29/systemd-primer-from-the-trenches.html.md

```diff
@@ -0,0 +1,514 @@
+                        └─local-fs.target @2.562s
+                          └─run-user-42.mount @4.614s
+                            └─local-fs-pre.target @964ms
+                              └─lvm2-monitor.service @321ms +442ms
+                                └─dm-event.socket @320ms
+                                  └─-.slice
+```
+
+Lets look at how long our network targets take to start:
+
+```# systemd-analyze critical-chain network.target```
+
+```bash
+icinga-ian ~]# systemd-analyze critical-chain network.target
+The time after the unit is active or started is printed after the "@" character.
+The time the unit takes to start is printed after the "+" character.
+
+network.target @2.618s
+└─network.service @2.403s +214ms
+  └─NetworkManager-wait-online.service @1.458s +941ms
+    └─NetworkManager.service @1.417s +40ms
+      └─network-pre.target @1.415s
+        └─firewalld.service @976ms +438ms
+          └─polkit.service @891ms +83ms
+            └─basic.target @885ms
+              └─paths.target @885ms
+                └─brandbot.path @885ms
+                  └─sysinit.target @881ms
+                    └─systemd-update-utmp.service @872ms +7ms
+                      └─auditd.service @702ms +168ms
+                        └─systemd-tmpfiles-setup.service @676ms +24ms
+                          └─rhel-import-state.service @653ms +22ms
+                            └─local-fs.target @652ms
+                              └─boot.mount @523ms +128ms
+                                └─local-fs-pre.target @522ms
+                                  └─lvm2-monitor.service @369ms +152ms
+                                    └─lvm2-lvmetad.service @397ms
+                                      └─lvm2-lvmetad.socket @368ms
+                                        └─-.slice
+```
+
+Lets go one step further and output the entire system heirachy. 
+
+``systemd-analyze dot``
+
+Now with this one you should output to nice res picture
+
+```systemd-analyze dot | dot -Tpng -o system-stuff.png```
+
+ 
+## Too many typings - I type enough everyday and a have bzillion servers
+
+*Pro Tip 2*
+
+I have a server which needs a service restarted or checked constantly. Running systemctl remotely will show me or 
+allow me to do this.
+
+```systemctl status sshd -H root@server.address.com```
+
+or
+
+```systemctl -H root@server.address status httpd```
+
+I might make an alias for it. Obviously this is a pretty useless example, if you’re having to manually do this for a 
+service/process you should fix the problem on the server. However for edge cases it can be quite handy. Use your imagination,
+we could use this for monitoring which takes a local ssh user found on all machines and pass this for some returned
+output to a monitoring server.
+
+## What relies on my service
+*Pro Tip 3*
+
+Figure out what targets/runlevel a target runs at. We will touch on 'WANTS' in next weeks 'UNIT' files journey.
+
+```systemctl show httpd -p “wants” multi-user.target```
```

Comment:
I'm not sure it makes sense to show this example here if you're not going to explain what the "wants" are till another post later.

Reply:
Yep, I was also tempted to pull this. Obviously needs more info behind> i did want to provide something rather nothing on targets. Let me see if I can show a better example